src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.Indenter.cs

```diff
@@ -71,48 +70,96 @@ public IndentationResult GetDesiredIndentation(FormattingOptions.IndentStyle ind
             {
                 // If the caller wants no indent, then we'll return an effective '0' indent.
                 if (indentStyle == FormattingOptions.IndentStyle.None)
+                    return default;
+
+                // If the user has explicitly set 'block' indentation, or they're in an inactive preprocessor region,
+                // then just do simple block indentation.
+                if (indentStyle == FormattingOptions.IndentStyle.Block ||
+                    _syntaxFacts.IsInInactiveRegion(Document.SyntaxTree, LineToBeIndented.Start, this.CancellationToken))
                 {
-                    return new IndentationResult(basePosition: 0, offset: 0);
+                    return GetDesiredBlockIndentation();
                 }
```

Comment:
extracted out the block-indentation logic.  also simplified things greatly by saying that if you're in an inactive-region (i.e. `#if false`) that you get block indentation.  This gets rid of a whole lot of logic to try to do weird things when inside something inactive.